File: remove-unused-imports.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Function to remove unused imports
function removeUnusedImports(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;
    
    const lines = content.split('\n');
    const newLines = [];
    
    for (let i = 0; i < lines.length; i++) {
      const line = lines[i];
      
      // Skip unused import lines
      if (line.includes('error TS6133') || line.includes('error TS6192')) {
        continue;
      }
      
      // Remove specific unused imports
      if (line.includes('memo') && line.includes('useCallback') && line.includes('import')) {
        // Keep only React import
        newLines.push("import React from 'react';");
        modified = true;
        continue;
      }
      
      if (line.includes('Navigation') && line.includes('import') && line.includes('unused')) {
        continue; // Skip unused Navigation imports
      }
      
      if (line.includes('Footer') && line.includes('import') && line.includes('unused')) {
        continue; // Skip unused Footer imports
      }
      
      if (line.includes('ErrorBoundary') && line.includes('import') && line.includes('unused')) {
        continue; // Skip unused ErrorBoundary imports
      }
      
      newLines.push(line);
    }
    
    if (modified) {
      content = newLines.join('\n');
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`Removed unused imports from: ${filePath}`);
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Fix specific files with known issues
function fixSpecificFiles() {
  const filesToFix = [
    'app/components/Navigation.tsx',
    'app/page.tsx',
    'app/page-new.tsx',
    'app/page-optimized.tsx',
    'app/5g-mobile-applications/page.tsx'
  ];
  
  let fixedCount = 0;
  
  for (const file of filesToFix) {
    const filePath = path.join(__dirname, file);
    if (fs.existsSync(filePath)) {
      if (removeUnusedImports(filePath)) {
        fixedCount++;
      }
    }
  }
  
  return fixedCount;
}

// Run the fix
console.log('Removing unused imports...');
const fixedCount = fixSpecificFiles();
console.log(`Fixed ${fixedCount} files with unused import issues.`);

File: remove_unused_imports.py
#!/usr/bin/env python3
"""
Script to remove unused imports from TSX files.
"""

import os
import re
from pathlib import Path

def remove_unused_imports_in_file(file_path):
    """Remove unused imports from a single file."""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Check if file has unused imports
        if 'ArrowRight' in content and '<ArrowRight' not in content:
            print(f"Removing unused ArrowRight import from: {file_path}")
            
            # Remove ArrowRight from lucide-react import
            content = re.sub(
                r'import \{\s*ArrowRight,\s*([^}]+)\s*\} from \'lucide-react\';',
                r'import { \1 } from \'lucide-react\';',
                content
            )
            
            # Remove ArrowRight from single import
            content = re.sub(
                r'import \{\s*ArrowRight\s*\} from \'lucide-react\';',
                '',
                content
            )
            
            # Clean up empty lines
            lines = content.split('\n')
            cleaned_lines = []
            prev_empty = False
            
            for line in lines:
                line = line.rstrip()
                if line.strip() == '':
                    if not prev_empty:
                        cleaned_lines.append('')
                    prev_empty = True
                else:
                    cleaned_lines.append(line)
                    prev_empty = False
            
            content = '\n'.join(cleaned_lines)
            
            # Write the fixed content back
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            
            return True
        
        return False
        
    except Exception as e:
        print(f"Error processing {file_path}: {e}")
        return False

def main():
    """Main function to remove unused imports."""
    workspace_root = Path('/workspace')
    app_dir = workspace_root / 'app'
    
    if not app_dir.exists():
        print("App directory not found!")
        return
    
    # Find all TSX files
    tsx_files = list(app_dir.rglob('*.tsx'))
    
    print(f"Found {len(tsx_files)} TSX files to check")
    
    fixed_count = 0
    for file_path in tsx_files:
        if remove_unused_imports_in_file(file_path):
            fixed_count += 1
    
    print(f"Removed unused imports from {fixed_count} files")

if __name__ == "__main__":
    main()